### PR TITLE
vk: use std::unordered_map in PipelineLayoutCache

### PR DESCRIPTION
**Pull request by poweifeng**
This is to address a weird map.find() miss that only happens on ARM in release mode.

### CODE REVIEW

**Review by pixelflinger — APPROVED** [review]

File: filament/backend/src/vulkan/caching/VulkanPipelineLayoutCache.h [hunk 1]
```diff
@@ -36,8 +36,8 @@ class VulkanPipelineLayoutCache {
     void terminate() noexcept;
 
     struct PushConstantKey {
-        uint8_t stage;// We have one set of push constant per shader stage (fragment, vertex, etc).
-        uint8_t size;
+        uint8_t stage = 0;// We have one set of push constant per shader stage (fragment, vertex, etc).
+        uint8_t size = 0;
```
Comment (pixelflinger) [inline filament/backend/src/vulkan/caching/VulkanPipelineLayoutCache.h:40]: couldn't the problem come from that?

**Review by bejado — APPROVED** [review]
LGTM, but if there's a bug with `tsl::robin_map`, we should investigate because we use that all over the place. In my experience, bugs have been related to padding bytes.

File: filament/backend/src/vulkan/caching/VulkanPipelineLayoutCache.h (hunk 1 above)
Comment (poweifeng) [reply filament/backend/src/vulkan/caching/VulkanPipelineLayoutCache.h:40]: that's what i thought. But just making this change does not fix the problem unfortunately.

**Comment by poweifeng** [comment]
> LGTM, but if there's a bug with `tsl::robin_map`, we should investigate because we use that all over the place. In my experience, bugs have been related to padding bytes.

Agreed.  Let me post to our chat more about this.  I experimented with various things, and the cause still eludes me.  AFAICT there are no padding issues in this key.

**Comment by poweifeng** [comment]
Internal bug for tracking has been created.  Landing this just to unblock usage on certain devcies.

**Pull request merged**